generator/internal/metadata_decorator_generator.cc

```diff
@@ -397,16 +397,21 @@ std::unique_ptr<::google::cloud::internal::AsyncStreamingReadRpc<
       continue;
     }
     if (IsStreamingWrite(method)) {
+      // Asynchronous streaming writes do not consume a request, typically the
+      // first `Write()` call contains any relevant data to "start" the stream.
+      // Thus, the decorator cannot add any routing instructions. The caller
+      // should initialize `context` with any such instructions. At this time,
+      // all callers are hand-crafted, so this is just a bit more work for
+      // library developers, ut a lot less than writing the routing instructions
+      // for all RPCs.
       auto const definition = absl::StrCat(
           R"""(
 std::unique_ptr<::google::cloud::internal::AsyncStreamingWriteRpc<
     $request_type$, $response_type$>>
 $metadata_class_name$::Async$method_name$(
     google::cloud::CompletionQueue const& cq,
     std::unique_ptr<grpc::ClientContext> context) {
-)""",
-          SetMetadataText(method, kPointer),
-          R"""(
+  SetMetadata(*context);
```

Comment:
hm, so the comment you wrote also applies to async streaming read-writes. It seems like we should either copy the comment, or have both use `SetMetadataText(method, kPointer)`. Then in that method we would say like:
```cc
std::string SetMetadataText(...) {
  std::string const context = ...;

  // blah blah blah
  // comment about client streaming
  // blah blah blah
  if (method.client_streaming()) return "  SetMetadata(" + context + ");";
  ...
}
```

Reply:
I chose to copy the comment